### DISCUSSION TITLE
Add `*_$type` dynamic Solr fields for non-date types

### DISCUSSION THREAD

**Discussion by frafra** [ckan/ckan#6593] [Ideas]
https://github.com/ckan/ckan/blob/36839d57d443bd189056bc83cf3e684c72a1f5d5/ckan/config/solr/schema.xml#L169

This is useful for people wanting to add a date field without having to change Solr configuration. It would make sense to use the same approach for other Sorl types, like `int`, `time`, `bool`.

We also have dynamic fields for multivalued strings and strings already.

What do you think about it?

**Comment by wardi** [comment]
I'm not a fan of Hungarian notation for metadata fields. These conventions leak into the API of our objects even if we don't want them and may break compatibility with existing schemas based on the field names a site chose.

Once we're using dynamic schemas in solr 8 we should be able to programmatically define the types of fields in solr so they can be generated based on e.g. a ckanext-scheming schema or similar.

**Reply by frafra** [reply]
That seems great :) I usually add the additional fields during the plugin initialization, but I guess there will be better ways to handle them with Solr 8.
